SMDT1.py

```diff
@@ -0,0 +1,168 @@
+from Bio import SeqIO
+from Bio.SeqUtils import gc_fraction
+from typing import Union
+import numpy as np
+import os
+import string
+
+COMP_BASES_DNA: dict = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A', 'a': 't',
+                        'g': 'c', 'c': 'g', 't': 'a'}
+COMP_BASES_RNA: dict = {'A': 'U', 'C': 'G', 'G': 'C', 'U': 'A', 'a': 'u',
+                        'g': 'c', 'c': 'g', 'u': 'a'}
+AA_MASSES: dict = {'A': 89, 'R': 174, 'N': 132, 'D': 133, 'C': 121, 'Q': 146, 'E': 147, 'Z': 147,
+                   'G': 75, 'H': 155, 'I': 131, 'L': 131, 'K': 146, 'M': 149, 'F': 165, 'P': 115, 'S': 105,
+                   'T': 119, 'W': 204, 'Y': 181, 'V': 117}
+
+class BiologicalSequence:
+    def __init__(self, seq):
+        self.seq = seq
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __str__(self):
+        return str(self.seq)
+
+    def __repr__(self):
+        return f"{self.seq}"
+
+    def __getitem__(self, key):
+        if isinstance(key, int):
+            item = str(self).__getitem__(key)
+            return item
+        elif isinstance(key, slice):
+            # Get the start, stop, and step from the slice
+            piece = [self[i] for i in range(*key.indices(len(self)))]
+            return ''.join(piece)
+    def check_alphabet(self):
+        return set(self.seq) <= set('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ')
```

Comment:
неплохо написано! но мне кажется что надо было использовать абстрактный класс. как я сделал это у себя:
```suggestion
from abc import ABC, abstractmethod


class BiologicalSequence(ABC):
    @abstractmethod
    def __init__(self, sequence):
        pass

    @abstractmethod
    def __len__(self):
        pass

    @abstractmethod
    def __getitem__(self, slc):
        pass

    @abstractmethod
    def __str__(self):
        pass

    @abstractmethod
    def __repr__(self):
        pass

    @abstractmethod
    def is_valid_alphabet(self):
        pass
```